packages/cli/commands/project/dev.js

```diff
@@ -104,12 +108,30 @@ exports.handler = async options => {
   let createNewSandbox = false;
   let createNewDeveloperTestAccount = false;
 
+  // The user is targeting an account type that we recommend developing on
   if (!targetProjectAccountId && defaultAccountIsRecommendedType) {
-    await confirmDefaultAccountIsTarget(accountConfig, hasPublicApps);
-    targetProjectAccountId = hasPublicApps
-      ? accountConfig.parentAccountId
-      : accountId;
     targetTestingAccountId = accountId;
+
+    await confirmDefaultAccountIsTarget(accountConfig, hasPublicApps);
+
+    if (hasPublicApps) {
+      // Exit if the user has not authed the parent account in the config
+      if (!getAccountConfig(accountConfig.parentAccountId)) {
+        logger.error(
+          i18n(`${i18nKey}.errors.parentAccountNotConfigured`, {
+            accountId: accountConfig.parentAccountId,
+            accountIdentifier: uiAccountDescription(targetTestingAccountId),
+            authCommand: uiCommandReference(
+              `hs auth --account=${accountConfig.parentAccountId}`
+            ),
+          })
+        );
+        process.exit(EXIT_CODES.ERROR);
+      }
+      targetProjectAccountId = accountConfig.parentAccountId;
+    } else {
+      targetProjectAccountId = accountId;
+    }
```

Comment:
I think this could be pulled into `lib/localDev`

Reply:
@camden11 just to clarify, which part do you think should be pulled in? All this logic is really doing is checking to see if the parent account exists in the config or not.

Reply:
I spent a little time trying to do this, but I couldn't get something I was happy with. There are so many args that need to be passed into the abstracted util that it ends up not feeling very valuable. I'm going to merge this as-is, but feel free to tweak this later on if you can think of a cleaner format for it!